src/app/features/data-import/data-import.service.ts

```diff
@@ -18,12 +18,47 @@ export class DataImportService {
     private snackBar: MatSnackBar
   ) {}
 
+    async validateCsvFile(file: File, entityType: string): Promise<boolean> {
+      const csvData = await readFile(file);
+      const parsedCsvFile = this.parseCsvFile(csvData);
+
+      // an empty csv file is not valid
+      if (parsedCsvFile.data.length === 0) {
+        // TODO: Either open a popup here which seems missplaced
+        // Better: Have validation results, so any component can handle the result
+        return false;
+      }
+
+      const record = parsedCsvFile.data[0];
+
+      // check all properties, if there is an _id, make sure it fits
+      for (const propertyName in record) {
+        if (propertyName !== "_id") {
+          continue
+        }
+
+        if (!record[propertyName].startsWith(entityType)) {
```

Comment:
This check is not working. I am getting `record[propertyName] === undefined` even when I have provided IDs. I would suggest you to write tests that ensure the code is correct (Test-Driven-Development!).

Reply:
Hm, my newly written tests did what I expected them to do in each case

Reply:
Ok did you try this manually then? Upload a CSV file with a entity type in the `_id` column?

Reply:
Yeah I actually did, well to the point the import would start which still requires the file to be validated.
I have a file with _id (and 122 records) and one without _id (and 1 record), maybe we can have a look at this together